Create neatsvor_x520ru_vacuum.yaml

Flegm opened the pull request.

make-all commented on custom_components/tuya_local/devices/neatsvor_x520ru_vacuum.yaml:
```diff
@@ -0,0 +1,153 @@
+        unit: min
+        class: measurement
+  - entity: select
+    name: Language
```
Use `translation_key: language` instead of name and icon.

make-all commented on custom_components/tuya_local/devices/neatsvor_x520ru_vacuum.yaml:
```diff
@@ -0,0 +1,153 @@
+        class: measurement
+  - entity: sensor
+    name: Last cleaning area
+    icon: mdi:floor-plan
```
Use `class: area` instead of the icon.

make-all commented on custom_components/tuya_local/devices/neatsvor_x520ru_vacuum.yaml:
```diff
@@ -0,0 +1,153 @@
+          - dps_val: "3"
+            value: "Cleaning along the walls"
+          - dps_val: "4"
+            value: "Recharging"
```
This should be "charging" to be interpreted correctly.

make-all commented on custom_components/tuya_local/devices/neatsvor_x520ru_vacuum.yaml:
```diff
@@ -0,0 +1,153 @@
+        name: status
+        mapping:
+          - dps_val: "0"
+            value: "Standby"
```
This needs to be all lower case to be picked up by the vacuum entity as a supported state.

make-all commented on custom_components/tuya_local/devices/neatsvor_x520ru_vacuum.yaml:
```diff
@@ -0,0 +1,153 @@
+          - dps_val: "4"
+            value: "Recharging"
+          - dps_val: "5"
+            value: "Charging"
```
I don't know what the difference is supposed to be between "Recharging" and "Charging". Is one supposed to be "charged"?

make-all commented on custom_components/tuya_local/devices/neatsvor_x520ru_vacuum.yaml:
```diff
@@ -0,0 +1,153 @@
+          - dps_val: "english"
+            value: "english"
+  - entity: select
+    name: Water step
```
Use `translation_key: mopping` instead of the name and icon here, and map the values to "low", "medium" and "high" (also "auto" and "off" are supported values if you later find it supports those).

Flegm commented:
I've fixed all the issues as requested:

1. Changed status values to lowercase (standby, charging) for proper vacuum state recognition
2. Replaced name and icon with translation_key: language for Language selector
3. Added class: area and removed icon for Last cleaning area sensor
4. Replaced name and icon with translation_key: mopping for water level control
5. Updated value mapping for mopping levels (medium instead of mid)

All changes are now aligned with project standards and translation files. Please let me know if anything else needs adjustment.
